.github/workflows/mac.yml

```
@@ -16,398 +16,403 @@ concurrency:
-        run: brew install coreutils ninja scons
-
-      - name: Build genai
+      - name: Build Samples (Release)
+        if: ${{ 'Release' == matrix.build-type }}
         run: |
-          source ${OV_INSTALL_DIR}/setupvars.sh
-          cmake -DCMAKE_BUILD_TYPE=Release -S ./ -B ./build/
-          cmake --build ./build/ --config Release --target py_openvino_genai --parallel --verbose
+          chmod +x ${{ env.OV_INSTALL_DIR }}/samples/cpp/build_samples.sh
+          ${{ env.OV_INSTALL_DIR }}/samples/cpp/build_samples.sh -i ${{ env.INSTALL_DIR }}
 
-      - name: Test bindings
+      - name: Build Samples (${{ matrix.build-type }})
+        if: ${{ 'Release' != matrix.build-type }}
         run: |
-          source ${OV_INSTALL_DIR}/setupvars.sh
-          python -m pip install ./thirdparty/openvino_tokenizers/[transformers] -r ./tests/python_tests/requirements.txt --find-links ${OV_INSTALL_DIR}/wheels
-          python -m pytest -v ./tests/python_tests/test_whisper_pipeline.py -k test_smoke
-        env:
-          PYTHONPATH: "./build/:$PYTHONPATH"
+          source ${{ env.OV_INSTALL_DIR }}/setupvars.sh
+          cmake -DCMAKE_BUILD_TYPE=${{ matrix.build-type }} -S ${{ env.OV_INSTALL_DIR }}/samples/cpp/ -B ${{ env.BUILD_DIR }}
+          cmake --build ${{ env.BUILD_DIR }} --config ${{ matrix.build-type }} --parallel $(nproc)
+          cmake --install ${{ env.BUILD_DIR }} --config ${{ matrix.build-type }} --component samples_bin --prefix ${{ env.INSTALL_DIR }}
 
-      - name: Test bindings (wheel)
-        run: |
-          source ${OV_INSTALL_DIR}/setupvars.sh
-          python -m pip install . --verbose --find-links ${OV_INSTALL_DIR}/wheels
-          python -c "from openvino_genai import LLMPipeline"
-          python -m pip install ./tools/who_what_benchmark --find-links ${OV_INSTALL_DIR}/wheels
-          python -m pytest -v ./tests/python_tests/test_whisper_pipeline.py -k "not test_smoke"
-
-  genai_package:
-    name: OpenVINO genai extension (install to OpenVINO package)
+      - name: Upload Samples Build Package
+        if: ${{ always() }}
+        uses: actions/upload-artifact@b4b15b8c7c6ac21ea08fcf65892d2ee8f75cf882 # v4.4.3
+        with:
+          name: genai_samples_${{ matrix.build-type }}
+          path: ${{ env.INSTALL_DIR }}
+          if-no-files-found: 'error'
+        
+  genai_tests_wheel:
+    name: Python (${{ matrix.test.name}}) Tests (wheel)
+    needs: [ openvino_download, genai_build_wheel ]
+    timeout-minutes: 60
     strategy:
+      fail-fast: false
       matrix:
-        build-type: [Release, Debug]
-    needs: [ openvino_download, openvino_build ]
-    if: |
-      always() &&
-      (needs.openvino_download.outputs.status == 'success' || needs.openvino_build.result == 'success')
-    timeout-minutes: 30
+        test:
+          - name: 'Whisper'
+            cmd: 'tests/python_tests/test_whisper_pipeline.py tests/python_tests/test_whisper_pipeline_static.py'
```

```suggestion
            cmd: 'tests/python_tests/test_whisper_pipeline.py'
```
as it requires NPU which is not available on macOS